bms-monitor.js

```diff
@@ -1,18 +1,62 @@
-const {expect} = require('chai');
-
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
-        console.log('Temperature is out of range!');
-        return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
-        console.log('Charge rate is out of range!');
-        return false;
-    }
-    return true;
-}
-
-expect(batteryIsOk(25, 70, 0.7)).to.be.true;
-expect(batteryIsOk(50, 85, 0)).to.be.false;
+const { Temperature, SOC, ChargeRate } = require('./batteryParameter');
+const { expect } = require('chai');
+const { BatteryState } = require('./batteryState');
+
+globalLanguage = 'en';
+
+let batteryState;
+batteryState = new BatteryState(
+	new Temperature(25),
+	new SOC(77),
+	new ChargeRate(0.7)
+);
+expect(batteryState.isBatteryStateOk).to.be.true;

```

Comment:
A number of combinations are possible temperature ( low breach, low warning, normal, high warning, high breach) along with SOC and chargerate. See if important ones can be covered via tests.

Reply:
All possible combination added for temperature, SOC, charge rate.
( low breach, low warning, normal, high warning, high breach)